Add SNI Support for imap check

e1mo opened the pull request:
As of Version of 2.3 of the monitoring-plugins[0], the check_imap
command also supports the `--sni` parameter.

This PR should only be merged, once icinga shipps with version 2.3 of
monitoring-plugins.

[^0]: https://github.com/monitoring-plugins/monitoring-plugins/releases/tag/v2.3

Al2Klimov commented on itl/command-plugins.conf:
```diff
@@ -900,6 +900,10 @@ object CheckCommand "imap" {
 			description = "Seconds before connection times out (default: 10)"
 		}
+		"--sni" = {
+			value = "$ssl_sni$"
```
Are you sure about this?

e1mo replied:
Oh sorry, my bad! Fixed in 92f4c4c39. However, I've not changed the description since it has a comparable description in the [monitoring-plugins documentation](https://www.monitoring-plugins.org/doc/man/check_imap.html): `SSL server_name`

Al2Klimov reviewed (changes requested):
Hello @e1mo and thank you for contributing!

Al2Klimov reviewed (changes requested):
Please squash your commits.

e1mo commented:
I have also added support for `simap`, `pop` and `spop` as those commands also support the sni argument. In addition, I also added the parameters to `doc/10-icinga-template-library.md`.

Al2Klimov commented:
## Tasks

* Verify whether it looks good to you
* Verify that the changed commands' plugins actually support the added parameters

Yonas-net reviewed (approved):
LGTM!

Al2Klimov commented:
@Yonas-net Do those `--sni`s take any values or are they just flags?

Yonas-net commented:
The parameter expects string values. You can also check it e.g. [here](https://www.monitoring-plugins.org/doc/man/check_simap.html).

Al2Klimov reviewed (changes requested):
Everything LGTM – ex. the description. It looks like the newly added parameters are booleans. Have a look at the SNI parameters of already existing commands.

Al2Klimov commented:
But I'm open for second opinions.

Al2Klimov commented:
@cla-bot check
